lib/puppet/file_system/windows.rb

```diff
@@ -3,6 +3,20 @@
 
 class Puppet::FileSystem::Windows < Puppet::FileSystem::Posix
 
+  def expanded_path_string(path)
+    string_path = path.to_s
+    # if no tildes, nothing to expand, return original string
+    # WARN: this will return a / on Windows if passed a /
+    return string_path if string_path !~ /~/
```

Comment:
this is slow in comparison to using `String#index` since it is a one character search. Using a regexp is overkill that creates result object(s), sets global variables and more...